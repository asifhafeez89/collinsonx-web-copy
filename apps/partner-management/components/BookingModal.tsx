import { Close } from '@collinsonx/design-system/assets/icons';
import { ActionIcon, Box, Modal } from '@collinsonx/design-system/core';
import { Booking } from '@collinsonx/utils';
import Details from './Details';

interface BookingModalProps {
  booking: Booking | null;
  onClickClose: () => void;
  children?: JSX.Element;
}
function BookingModal({ booking, onClickClose, children }: BookingModalProps) {
  return (
    <Modal
      opened={booking !== null}
      withCloseButton={false}
      onClose={onClickClose}
      padding={0}
      size={712}
    >
      <ActionIcon
        color="dark.6"
        onClick={onClickClose}
        sx={{
          position: 'absolute',
          top: 40,
          right: 40,
        }}
      >
        <Close w={24} h={24} />
      </ActionIcon>
      <Box p={40} pt={80}>
        {booking ? <Details booking={booking}>{children}</Details> : null}
      </Box>
    </Modal>
  );
}

export default BookingModal;
