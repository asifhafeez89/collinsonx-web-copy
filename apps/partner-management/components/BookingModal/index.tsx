import { Close } from '@collinsonx/design-system/assets/icons';
import { ActionIcon, Box, Modal } from '@collinsonx/design-system/core';
import { Booking } from '@collinsonx/utils';
import Details from '../Details';
import classes from './BookingModal.module.css';

interface BookingModalProps {
  booking: Booking | null;
  onClickClose: () => void;
  children?: JSX.Element;
}
function BookingModal({ booking, onClickClose, children }: BookingModalProps) {
  return (
    <Modal
      opened={booking !== null}
      withCloseButton={false}
      onClose={onClickClose}
      padding={0}
      size={712}
    >
      <ActionIcon
        color="dark.6"
        onClick={onClickClose}
        classNames={{ root: classes.closeIcon }}
      >
        <Close w={24} h={24} />
      </ActionIcon>
      <Box p={40} pt={80}>
        {booking ? <Details booking={booking}>{children}</Details> : null}
      </Box>
    </Modal>
  );
}

export default BookingModal;
