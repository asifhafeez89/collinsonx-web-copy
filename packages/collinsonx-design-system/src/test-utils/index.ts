import userEvent from '@testing-library/user-event';

export * from '@testing-library/react';
export { userEvent };
export { default as render, Provider } from './render';
