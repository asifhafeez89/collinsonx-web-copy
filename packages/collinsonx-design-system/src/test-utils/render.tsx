import { RenderOptions, render } from '@testing-library/react';
import { MantineProvider } from '@mantine/core';
import { ReactElement, ReactNode } from 'react';

export const Provider = ({ children }: { children: ReactNode }) => (
  <MantineProvider>{children}</MantineProvider>
);

type Render = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>
) => ReturnType<typeof render>;

const customRender: Render = (ui, options) =>
  render(ui, {
    wrapper: Provider as unknown as RenderOptions['wrapper'],
    ...options,
  });

export default customRender;
