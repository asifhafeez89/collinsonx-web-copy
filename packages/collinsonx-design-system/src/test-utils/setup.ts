import userEvent from '@testing-library/user-event';
import render from './render';

export default function (jsx: Parameters<typeof render>[0]) {
  return {
    user: userEvent.setup(),
    ...render(jsx),
  };
}
