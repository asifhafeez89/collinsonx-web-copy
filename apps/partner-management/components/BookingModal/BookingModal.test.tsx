import BookingModal from '.';
import '@testing-library/jest-dom/extend-expect';
import { render } from '@collinsonx/design-system/test-utils';

describe('<BookingModal />', () => {
  it('should render', () => {
    const component = render(
      <BookingModal onClickClose={() => {}} booking={null} />
    );

    expect(component.getByTestId('modal')).toBeInTheDocument();
  });
});
