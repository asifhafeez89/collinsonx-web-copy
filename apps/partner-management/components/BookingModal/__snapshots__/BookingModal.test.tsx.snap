// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BookingModal /> should render 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <style
        data-mantine-styles="classes"
      >
        @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
      </style>
    </div>
    <div
      data-portal="true"
    >
      <div>
        <div
          class="m-9df02822 mantine-Modal-root"
          style="--modal-size: calc(44.5rem * var(--mantine-scale)); --modal-y-offset: 5dvh; --mb-z-index: 200; --mb-padding: 0;"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <style
      data-mantine-styles="classes"
    >
      @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
    </style>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
